homeassistant/components/mqtt/__init__.py

```diff
@@ -594,4 +598,8 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if subscriptions := mqtt_client.subscriptions:
         mqtt_data.subscriptions_to_restore = subscriptions
 
+    # Unregister open config issues
+    while open_config_issues := mqtt_data.open_config_issues:
```

Comment:
`mqtt_data.open_config_issues` is and object (`set[str]`)set thought a `dataclass` using a default factory.
The intention is pop an item (containing an issue_id) and remove the registered issue.

Reply:
I was just thinking that 

```
            while open_config_issues := mqtt_data.open_config_issues:
                async_delete_issue(hass, DOMAIN, open_config_issues.pop())
```

could be a function since you do it in two places

Reply:
Ah, okay, now I see what you mean. Good point.

Reply:
done